modules/transactions/src/main/java/org/apache/ignite/internal/tx/ActiveTxMinimumBeginTimeProvider.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.tx;
+
+import org.apache.ignite.internal.hlc.HybridTimestamp;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Provides the minimum begin time among all active RW transactions started locally.
+ */
+@SuppressWarnings("InterfaceMayBeAnnotatedFunctional")
+public interface ActiveTxMinimumBeginTimeProvider {
+    /** Returns the minimum begin time among all active RW transactions started locally. */
+    @Nullable HybridTimestamp minimumBeginTime();
```

Comment:
BTW, curious what time are you going to use as global minimum in case of zero pending transactions cluster wide?

Reply:
we use timestamp obtained just BEFORE computing minimum tx time.

```java
    private long computeLocalMinTxBeginTime() {
        // To help compaction progress when there are no active RW transactions,
        // it is safe to return the current time as minimum tx begin time.
        HybridTimestamp beforeDetermineTxTime = clockService.now();

        return Objects.requireNonNullElse(
                localActiveRwTxMinBeginTimeProvider.minimumBeginTime(),
                beforeDetermineTxTime
        ).longValue();
    }
```

Reply:
>         // it is safe to return the current time as minimum tx begin time.
>         HybridTimestamp beforeDetermineTxTime = clockService.now();
I'm not sure that it's safe.
I mean that there should be some gap between starting the transaction (and thus assigning some timestamp as tx.startTime) and registering it in `org.apache.ignite.internal.tx.impl.TxManagerImpl#localRwTxCounter`. 
The atomicity is achieved by using `inUpdateRwTxCountLock`
```
        return localRwTxCounter.inUpdateRwTxCountLock(() -> {
            HybridTimestamp beginTs = clockService.now();

            localRwTxCounter.incrementRwTxCount(beginTs);

            return beginTs;
        });
```
And because you don't use the lock in computeLocalMinTxBeginTime the race is possible.